apt/message.go

```diff
@@ -0,0 +1,258 @@
+package apt
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// AptMessage represents a single RFC822 Apt message.
+type AptMessage struct {
```

Comment:
Does it make sense to have different sub-types for different messages? that way, code would be implicit and fields would be strongly typed. 
Something to consider, might be an overkill.